marklogic/src/main/scala/quasar/physical/marklogic/fs/package.scala

```diff
@@ -16,24 +16,59 @@
 
 package quasar.physical.marklogic
 
+import quasar.Predef._
+import quasar.Data
+import quasar.effect.{KeyValueStore, MonotonicSeq, Read}
+import quasar.fp._
+import quasar.fp.free._
 import quasar.fs._
-import quasar.fs.mount.FileSystemDef, FileSystemDef.DefErrT
+import quasar.fs.mount.{ConnectionUri, FileSystemDef}, FileSystemDef.DefErrT
+
+import com.marklogic.client._
+import com.marklogic.xcc._
+import java.net.URI
 
 import scalaz._, Scalaz._
 import scalaz.concurrent.Task
+import scalaz.stream.Process
 
 package object fs {
   val FsType = FileSystemType("marklogic")
 
-  def definition[S[_]](implicit S0: Task :<: S, S1: PhysErr :<: S)
-    : FileSystemDef[Free[S, ?]] = FileSystemDef.fromPF {
+  type Eff0[A] = Coproduct[
+                    KeyValueStore[ReadFile.ReadHandle, Process[Task, Vector[Data]], ?],
+                    KeyValueStore[WriteFile.WriteHandle, Unit, ?],
+                    A]
+  type Eff1[A] = Coproduct[MonotonicSeq, Eff0, A]
+  type Eff2[A] = Coproduct[Read[Client, ?], Eff1, A]
+  type Eff[A]  = Coproduct[Task, Eff2, A]
+
+  def inter(uri: ConnectionUri): Task[Eff ~> Task] = {
+    def dbClient: DatabaseClient = DatabaseClientFactory.newClient("localhost", 8000)
```

Comment:
Seems like we should be using the passed in `ConnectionUri` here, no? Also, for now I guess we can assume that XCC is being provided via the rest api so we can use the same port, but if we need both I'm not sure that will always be the case, may need to allow configuration of that via query params (i.e. `xccPort=1234&restPort=6789` or something) unless we're able to abandon rest completely and perform all operations via XCC.